src/proxy-ui-api/src/main/java/org/niis/xroad/restapi/openapi/ClientsApiController.java

```diff
@@ -336,9 +341,25 @@ private ClientType getClientType(String encodedId) {
                 // deviation data (errorcode + warnings) copied
                 throw new ConflictException(e);
             }
-
         } else if (serviceDescription.getType() == ServiceType.OPENAPI3) {
-            return new ResponseEntity<>(HttpStatus.NOT_IMPLEMENTED);
+            try {
+                addedServiceDescriptionType = serviceDescriptionService.addOpenapi3ServiceDescription(clientId, url,
+                        restServiceCode, ignoreWarnings);
```

Comment:
`restServiceCode` might be null here which causes problems later. Needs a null check and proper exception (BadRequestException)